src/Mesh/babylon.buffer.ts

```diff
@@ -123,13 +123,13 @@
          * @param vertexCount the vertex count (optional)
          * @param useBytes set to true if the offset is in bytes
          */
-        public updateDirectly(data: DataArray, offset: number, vertexCount?: number, useBytes = false): void {
+        public updateDirectly(data: DataArray, offset: number, vertexCount?: number, useBytes: boolean = false): void {
             if (!this._buffer) {
                 return;
             }
 
             if (this._updatable) { // update buffer
-                this._engine.updateDynamicVertexBuffer(this._buffer, data, useBytes ? offset : offset * 4, (vertexCount ? vertexCount * this.byteStride : undefined));
+                this._engine.updateDynamicVertexBuffer(this._buffer, data, useBytes ? offset : offset * this.byteStride, (vertexCount ? vertexCount * this.byteStride : undefined));

```

Comment:
I don't believe this is correct. If `useBytes` is false, then it should behave like before where `offset` is in float32 units which is a multiple of 4. `byteStride` has nothing to do with it.

Reply:
Shouldn't it be the same  for `vertexCount` then ?

Reply:
That's right.I missed this one

Reply:
For vertex count we are good as the bytestride is the stride in byte of the entire vertex

Reply:
So 4 for both. (I guess I was assuming it may be different, but actually it all ends in an array of Float32)

Reply:
Actually only one not both (vertex count remains unchanged)

Reply:
Could `Float32Array.BYTES_PER_ELEMENT` be used in that case ? Unless I am wrong it's exactly what the 4 here means. (It helps understand better I think).

Reply:
Works for me

Reply:
Perhaps update the other 4s in the file to this constant?